src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -44,7 +46,7 @@ private sealed class LatestDiagnosticsForSpanGetter
             private readonly IEnumerable<StateSet> _stateSets;
             private readonly CompilationWithAnalyzers? _compilationWithAnalyzers;
 
-            private readonly TextSpan _range;
+            private readonly TextSpan? _range;
```

Comment:
ok.  i tried to trace down everywhere this could reach, and this does appear to have the semantics we want.  so allowing the caller to pass in null seems like the simplest and cheapest approach.